src/GraphQLParser/AST/GraphQLScalarValue.cs

```diff
@@ -1,102 +0,0 @@
-using System.Diagnostics;
-
-namespace GraphQLParser.AST
-{
-    /// <summary>
-    /// Scalar nodes represent primitive leaf values in a GraphQL document.
-    /// <br/>
-    /// There are 6 kinds of scalar nodes:
-    /// <br/>
-    /// <see cref="ASTNodeKind.StringValue">String</see>
-    /// <br/>
-    /// <see cref="ASTNodeKind.BooleanValue">Boolean</see>
-    /// <br/>
-    /// <see cref="ASTNodeKind.IntValue">Int</see>
-    /// <br/>
-    /// <see cref="ASTNodeKind.FloatValue">Float</see>
-    /// <br/>
-    /// <see cref="ASTNodeKind.EnumValue">Enumeration</see>
-    /// <br/>
-    /// <see cref="ASTNodeKind.NullValue">Null</see>
-    /// </summary>
-    [DebuggerDisplay("{Value}")]
-    public class GraphQLScalarValue : GraphQLValue
```

Comment:
Bad design here, misleading naming. I removed this node. Also now all child nodes don't store `ASTNodeKind` -> less memory.